### PR TITLE
Media gallery synchronization does not remove obsolete images from database

### LINKED ISSUE #1261: Media gallery synchronization does not remove obsolete images from database
### Preconditions (*)
1. Magento 2.4-develop with Adobe Stock Integration 1.1-develop is installed
2. Enhanced Media Gallery is enabled

### Steps to reproduce (*)
1. Add an image to pub/media directory
2. Run `bin/magento media-gallery:sync`
3. Open Enhanced Media Gallery and check that added image is present
4. Remove the image from pub/media directory (via filesystem)
5. Run `bin/magento media-gallery:sync`
6. Open the Enhanced Media Gallery

### Expected result (*)
The removed image is not present in the media gallery

### Actual result (*)
The removed image is present in the media gallery

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Description (*)

Closes #1261

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#1261: Media gallery synchronization does not remove obsolete images from database
2. ...

### Manual testing scenarios (*)

1. ...
2. ...

### CODE REVIEW

File: MediaGallerySynchronization/Model/Synchronize.php [hunk 1]
```diff
@@ -101,6 +113,8 @@ public function execute(): void
             }
         }
 
+        $this->resolveNonExistedAssets->execute($assetsPaths);
```
Comment (sivaschenko) [inline MediaGallerySynchronization/Model/Synchronize.php:116]: Can you please utilize `\Magento\MediaGalleryApi\Api\DeleteAssetsByPathsInterface` for removing assets.

I.e. load assets to be removed with custom SQL query and than remove them using the call to API

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @engcom-Charlie ! Please take a look at my comment

**Review by sivaschenko — APPROVED** [review]

File: MediaGallerySynchronization/Model/Synchronize.php [hunk 2]
```diff
@@ -144,4 +158,15 @@ private function getMediaDirectory(): Read
...
+
+    /**
+     * Return asset's relative path
+     *
+     * @param string $assetPath
+     * @return string
+     */
+    private function getRelativePath(string $assetPath): string
+    {
+        return substr($assetPath, strlen($this->getMediaDirectory()->getAbsolutePath()));
```
Comment (sivaschenko) [inline MediaGallerySynchronization/Model/Synchronize.php]: `getRelativePath` method of directory instance can be used to retrieve relative path
```suggestion
        return $this->getMediaDirectory()->getRelativePath($assetPath);
```

File: MediaGallerySynchronization/Model/Synchronize.php [hunk 3]
```diff
@@ -89,6 +98,9 @@ public function execute(): void
                 continue;
             }
 
+            $relativePath = $this->getRelativePath($path);
+            $assetsPaths[] = $relativePath === $item->getFilename() ? '/' . $relativePath : $relativePath;
```
Comment (sivaschenko) [inline MediaGallerySynchronization/Model/Synchronize.php]: I think it would be good to move this code to `getRelativePath` private method

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for updates @engcom-Charlie ! Please see my review suggestions

**Review by sivaschenko — APPROVED** [review]

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

**Comment by sivaschenko** [comment]
@magento run Unit tests

**Comment by sivaschenko** [comment]
@magento run Functional Tests EE

**Pull request merged**